devices.js

```diff
@@ -14771,7 +14789,14 @@ const devices = [
 
 module.exports = devices.map((device) => {
     if (device.extend) {
+        let deviceMeta = null;
+        if (device.extend.hasOwnProperty('meta') && device.hasOwnProperty('meta')) {
+            deviceMeta = Object.assign({}, device.extend.meta, device.meta);
+        }
         device = Object.assign({}, device.extend, device);
+        if (deviceMeta !== null) {
+            device.meta = deviceMeta;
+        }
         delete device.extend;
```

Comment:
We should check if the configure is not overridden here. 


```js
if (device.extend.hasOwnProperty('configure') && device.hasOwnProperty('configure')) {
  assert.fail(`'${device.model}' has configure in extend and device, this is not allowed`);
}
```

If `npm tst` passes you are good!

Reply:
There is one device, I already had it on my too look at list.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/de0abee0297f9fe95c1bfaa68562378ea8ad1f7a/devices.js#L10982

I'll give this another pass later this week.

Reply:
I tried adding the check, it fails... but then complains about assert not being defined.

```
[hyperon :: sjorge][~/Documents/repos/zigbee-herdsman-converters][ctminmax ✗]
[■]$ npm tst

> zigbee-herdsman-converters@14.0.1 test /Users/sjorge/Documents/repos/zigbee-herdsman-converters
> jest test

 PASS  test/otaCommon.test.js
 FAIL  test/index.test.js
  ● Test suite failed to run

    ReferenceError: assert is not defined

      14791 |     if (device.extend) {
      14792 |         if (device.extend.hasOwnProperty('configure') && device.hasOwnProperty('configure')) {
    > 14793 |             assert.fail(`'${device.model}' has configure in extend and device, this is not allowed`);
            |             ^
      14794 |         }
      14795 |
      14796 |         let deviceMeta = null;

      at map (devices.js:14793:13)
          at Array.map (<anonymous>)
      at Object.<anonymous> (devices.js:14790:26)
      at Object.<anonymous> (index.js:3:17)
      at Object.<anonymous> (test/index.test.js:1:1)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       8 passed, 8 total
Snapshots:   0 total
Time:        2.11 s
Ran all test suites matching /test/i.
npm ERR! Test failed.  See above for more details.
[hyperon :: sjorge][~/Documents/repos/zigbee-herdsman-converters][ctminmax ✗]
[■]$ git diff
diff --git a/devices.js b/devices.js
index 8bce79e..10c273a 100755
--- a/devices.js
+++ b/devices.js
@@ -14789,14 +14789,21 @@ const devices = [

 module.exports = devices.map((device) => {
     if (device.extend) {
+        if (device.extend.hasOwnProperty('configure') && device.hasOwnProperty('configure')) {
+            assert.fail(`'${device.model}' has configure in extend and device, this is not allowed`);
+        }
+
         let deviceMeta = null;
         if (device.extend.hasOwnProperty('meta') && device.hasOwnProperty('meta')) {
             deviceMeta = Object.assign({}, device.extend.meta, device.meta);
         }
+
         device = Object.assign({}, device.extend, device);
+
         if (deviceMeta !== null) {
             device.meta = deviceMeta;
         }
+
         delete device.extend;
     }
```

Reply:
```const assert = require('assert');``` might fix it, let me try again.

Reply:
@Koenkk I don't think the one with a conflict needs the configure on the device, I'll drop it but if you think it does need it. We can copy the fromZigbee/toZigbee converter list and add the read calls to the configure instead.